File: TP2/Dizainier/Dizainier/ViewController.m
//
//  ViewController.m
//  Dizainier
//
//  Created by m2sar on 26/09/2015.
//  Copyright (c) 2015 m2sar. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    [self updateAll:0];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


- (void)updateAll:(int)newValue {
    _myStepper.value = newValue;
    _mySlider.value = newValue;
    _mySegmentedUnites.selectedSegmentIndex = newValue % 10;
    _mySedmentedDizaines.selectedSegmentIndex = newValue / 10;
    [self updateLabel:newValue];
}

- (void)updateLabel:(int)newValue {
    _myLabel.text = [NSString stringWithFormat:((_mySwitch.isOn) ? @"%X" : @"%d"), newValue];
    _myLabel.textColor = (newValue != 42) ? [UIColor blackColor] : [UIColor redColor];
}

- (IBAction)onClickMyStepper:(UIStepper *)sender {
    [self updateAll:_myStepper.value];
}

- (IBAction)onClickMySwitch:(UISwitch *)sender {
    [self updateLabel:_myStepper.value];
}

- (IBAction)onClickSegmentedCtrl:(UISegmentedControl *)sender {
    [self updateAll:((int)_mySedmentedDizaines.selectedSegmentIndex*10)+(int)_mySegmentedUnites.selectedSegmentIndex];
}

- (IBAction)onClickSlider:(UISlider *)sender {
    [self updateAll:(int)_mySlider.value];
}

- (IBAction)onTouchUpRaZ:(UIButton *)sender {
    [self updateAll:0];
}
@end

File: TP3/DizainierSwift/DizainierSwift/ViewController.swift
//
//  ViewController.swift
//  DizainierSwift
//
//  Created by m2sar on 01/10/2015.
//  Copyright (c) 2015 m2sar. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var myStepper: UIStepper!
    @IBOutlet weak var mySwitch: UISwitch!
    @IBOutlet weak var mySCDizaines: UISegmentedControl!
    @IBOutlet weak var mySCUnites: UISegmentedControl!
    @IBOutlet weak var myLabel: UILabel!
    @IBOutlet weak var mySlider: UISlider!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        self.updateAll(0);
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func updateAll(newValue: Int) {
        self.myStepper.value = Double(newValue);
        self.mySlider.value  = Float(newValue);
        self.mySCDizaines.selectedSegmentIndex = newValue / 10;
        self.mySCUnites.selectedSegmentIndex   = newValue % 10;
        self.updateLabel(newValue);
    }
    
    func updateLabel(newValue: Int) {
        self.myLabel.text = (self.mySwitch.on) ? String(format:"%X", newValue) : String(format:"%d", newValue);
        self.myLabel.textColor = (newValue != 42 ) ? UIColor.blackColor() : UIColor.redColor();
    }
    
    @IBAction func onChangeStepper(sender: AnyObject) {
        self.updateAll(Int(self.myStepper.value));
    }

    @IBAction func onChangeSwitch(sender: AnyObject) {
        self.updateLabel(Int(self.myStepper.value));
    }
    
    @IBAction func onChangeSC(sender: AnyObject) {
        self.updateAll(Int(self.mySCDizaines.selectedSegmentIndex * 10 + self.mySCUnites.selectedSegmentIndex));
    }
    
    @IBAction func onChangeSlider(sender: UISlider) {
        self.updateAll(Int(self.mySlider.value));
    }
    
    @IBAction func onClickRaZ(sender: AnyObject) {
        self.updateAll(0);
    }
    
}

